### PR TITLE
Don't store RLPs in valueNodes (except tests).

### PR DESCRIPTION
**Pull request by mandrigin**
fixes: #126

### CODE REVIEW

File: trie/proof_generator_test.go [hunk 1]
```diff
@@ -182,7 +182,8 @@ func TestSerialiseBlockWitness(t *testing.T) {
 	if err := bwb.WriteTo(&b); err != nil {
 		t.Errorf("Could not make block witness: %v", err)
 	}
-	expected := common.FromHex("0xa76862616c616e6365730065636f64657300666861736865731822646b65797300666e6f6e63657300697374727563747572650b6676616c75657300582023181a62d35fe01562158be610f84e047f99f5e74d896da21682d925964ece3a0601024704010402040304")
+
+	expected := common.FromHex("0xa76862616c616e6365730065636f64657300666861736865731822646b65797300666e6f6e63657300697374727563747572650b6676616c756573005820858f70a4b1e6aa71a7edc574d2ca946495a038aa37ce13dc7b7ed15661a6ff2f0601024704010402040304")
```
Comment (AlexeyAkhunov) [inline trie/proof_generator_test.go]: Why did this value change?

Comment (mandrigin) [reply trie/proof_generator_test.go]: we no longer treat `[]byte("val1")` and `[]byte("val2")` in `tr.Update(...)` calls (lines 174-175) as RLP-encoded values, so we add a double prefix to them when hashing, so they produce different hash values.

File: trie/hasher.go [hunk 2]
```diff
@@ -155,13 +155,64 @@ func generateByteArrayLen(buffer []byte, pos int, l int) int {
...
 	if l < 55 {
 		// After first wrapping, the length will be l + 1 < 56
 		buffer[pos] = byte(128 + l + 1)
 		pos++
 		buffer[pos] = byte(128 + l)
 		pos++
+	} else if l < 56 {
+		buffer[pos] = byte(183 + 1)
+		pos++
+		buffer[pos] = byte(l + 1)
```
[the remark this answers is not in the document]
Comment (mandrigin) [reply trie/hasher.go]: will clarify in https://github.com/ledgerwatch/turbo-geth/pull/171 if you don't mind :)

File: trie/tapes.go [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+	Next() (*big.Int, error)
+}
+
+// HashTape is an abstraction for an input table that allows reading 32-byte hashes (common.Hash) sequentially
+// To be used for intermediate hashes in the Patricia Merkle tree
+type HashTape interface {
+	Next() (common.Hash, error)
+}
+
+// FIXME: comments here
```
[the remark this answers is not in the document]
Comment (mandrigin) [reply trie/tapes.go]: yep, will do in https://github.com/ledgerwatch/turbo-geth/pull/171

**Pull request merged**